internal/generator/templates/decode.go.tmpl

```diff
@@ -69,6 +69,7 @@
             {{- if $native.IsMarshaler }}
                 {{- $element = "*elem" }}
             {{- end }}
+            {{ $assign_str_lvalue }} = make([]{{ if $native.RequiresCast }}{{ goType $scope $field.Type }}{{ else }}{{ goType $scope $native.Type }}{{ end }}, 0, len({{ $field.Name }}ArrayProperties.RawArray))
```

Comment:
```suggestion
            {{ $assign_str_lvalue }} = make([]{{ goType $scope $field.Type }}, 0, len({{ $field.Name }}ArrayProperties.RawArray))
```
I wonder if the if/else for the native (fundamental) type vs. the field type is really needed. Can we have a quick test if all the tests pass if we use the field type only?

Reply:
Using your suggested line yields green test (via `bazel test --test_output=all //... `)
My personal "test set" is also generated properly.

Do you want me to commit this change?

Reply:
Please do!